### ISSUE TITLE
Github Mirrors and Wiki

### ISSUE THREAD

**Issue by IzzySoft** [go-gitea/gitea#2309]
- Gitea version (or commit ref): 1.1.3
- Git version: 2.7.4
- Operating system: Bananian (Debian on ARM)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
I've set up a couple of mirrors from Github. If they have a Wiki when the mirror is created, it comes along (I didn't check yet if such a wiki is updated later). But if the Wiki is added later, future syncs don't seem to pick it up. Is there any step I've missed, or anything I can do to have Gitea picking it up – apart from deleting the mirror and creating a new one?

**Comment by lunny** [comment]
Maybe a bug I think.

**Comment by IzzySoft** [comment]
Seems like. I've meanwhile verified two things:

* if the wiki was active to begin with, it is mirrored correctly (so updates are pulled in)
* simply creating a mirror by hand doesn't help, it is not picked up (and I could not figure where to make the connection)

Sure, only being a "read-only mirror" one could simply drop and recreate it, but (especially for large repositories and on slow connections) this would be an ugly work-around :wink:

**Comment by bkcsoft** [comment]
Not a bug, since we only verify that the remote repo (the wiki in this case) exists when we start the import. Enhancement though 🙂 

Basically, show a "try import wiki"-button IF the repo is a mirror AND does not have a wiki setup.  
Maybe even have that button beside the "Create new wiki"-button on the wiki-page 🤔

**Comment by IzzySoft** [comment]
Bug, enhancement, feature … call it what you like as long as it works :grin:

And yes, such a button would be a good start. Though I think it could not hurt to automatically check for the existence of a wiki on each sync run. Gitea doesn't ask whether to look for it on the initial run either (and that's very OK this way), so I wouldn't expect I need to do so later. Instead of that button, I'd rather suggest a checkbox (on the first import screen and then later on the config screen of that repo) on whether I'm interested in the wiki. So one can always change one's mind.

**Comment by bkcsoft** [comment]
> Though I think it could not hurt to automatically check for the existence of a wiki on each sync run

Actually it could. Polling invalid repos on GitHub will get you rate-limited 🙁

**Comment by bkcsoft** [comment]
> Instead of that button, I'd rather suggest a checkbox (on the first import screen and then later on the config screen of that repo) on whether I'm interested in the wiki. So one can always change one's mind.

Sounds like a good idea though 🙂

**Comment by IzzySoft** [comment]
> Actually it could. Polling invalid repos on GitHub will get you rate-limited

Oh, wasn't aware of that. Which means it would need a kind of flag like "already checked today, there is no wiki – check again tomorrow" or the like, to prevent that. Until a "perfect approach" is found, that button would be helpful already (and not necessarily obsolete thereafter).

**Comment by IzzySoft** [comment]
Half a year later (I'm running Gitea 1.3.2 now): Did I miss the button, or did you miss adding it? I meanwhile lost track of which repos are simply lacking wikis on both ends, or which I have to trigger for sync. And there are several repos in my installation where I know I have to initiate the mirror – but apart from dropping the mirror entirely and recreating it, I still have no idea how to get it running.

**TL;DR:** any progress on this?

**Comment by bkcsoft** [comment]
No one is working on it. Feel free to contribute. People mainly work on bugs that they want/need to fix :/

**Comment by IzzySoft** [comment]
@bkcsoft I'd feel free if time would permit – and if I'd be a GO programmer. Got a bunch of other projects to care for, and I'm not a GO programmer – so unfortunately I can't.

**Comment by IzzySoft** [comment]
Oh dear bot, have mercy with the devs and grant them some more time 🤣 

(sorry, couldn't resist)

**Comment by linkp** [comment]
If this enhancement request ever does find it's way in as a feature, it would be nice to have an option to not import a wiki as part the same setting during mirror creation.

I can only find an option to disable the wiki after the mirror is created and I have a GithHub mirror that is failing after logging: [W] Clone wiki: exit status 128 - fatal: Remote branch master not found in upstream origin

I would love to have an option to test mirroring with the wiki disabled to see if that resolves the early termination of the mirroring process.

**Comment by lesinigo** [comment]
I'd add that it would be cool to be able to decide if the wiki is to be mirrored at all or not.

Use case: I want to mirror a _git repository_, but want to have my own wiki for my own notes / documentation.

Possible implementation:
* when creating the migration / mirror, ask the user if the Wiki should be migrated / mirrored
* if it's a migration, then everything goes on as usual
* if it is a mirror with mirrored wiki, it will stay as it is without further options
* if it is a mirror without mirrored wiki, allow to turn on Build-In or External wiki like on regular projects (and keep ignoring the remote wiki forever)

If the user changes his mind and wants to mirror the wiki when it didn't choose so (or the other way around), simply drop the project and re-create it.

**Comment by wohenbushuang** [comment]
> Use case: I want to mirror a git repository, but want to have my own wiki for my own notes / documentation.

Same issue when I found that I cannot create my notes to a mirror repository, even if I re-open/close the wiki function in repository settings, it won't show any button to create a new wiki page in the wiki tab
